hub/auto/unstructured/kaggle.py

```diff
@@ -0,0 +1,62 @@
+from hub.auto.unstructured.image_classification import ImageClassification
+import hub
+import glob
+import os
+from hub.util.exceptions import (
+    ExternalCommandError,
+    KaggleMissingCredentialsError,
+    KaggleDatasetAlreadyDownloadedError,
+)
+from hub.constants import ENV_KAGGLE_KEY, ENV_KAGGLE_USERNAME
+
+
+def _exec_command(command):
+    out = os.system(command)
+    if out != 0:
+        raise ExternalCommandError(command, out)
+
+
+def _set_environment_credentials_if_none(kaggle_credentials: dict = {}):
+    if ENV_KAGGLE_USERNAME not in os.environ:
+        username = kaggle_credentials.get("username", None)
+        os.environ[ENV_KAGGLE_USERNAME] = username
+        if not username:
+            raise KaggleMissingCredentialsError(ENV_KAGGLE_USERNAME)
+    if ENV_KAGGLE_KEY not in os.environ:
+        key = kaggle_credentials.get("key", None)
+        os.environ[ENV_KAGGLE_KEY] = key
```

Comment:
`if not  key` check should be done prior to setting the env var, as env vars can only be strings, not None.